libraries/botbuilder-adapter-twitter/src/twitter.ts

```diff
@@ -0,0 +1,195 @@
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    public async deleteActivity(context: TurnContext, reference: Partial<ConversationReference>): Promise<void> {
+        throw new Error('Method not supported by Twitter API.');
+    }
+
+    public async continueConversation(reference: Partial<ConversationReference>, logic: (context: TurnContext) => Promise<void>): Promise<void> {
+        const request: Partial<Activity> = TurnContext.applyConversationReference(
+            { type: 'event', name: 'continueConversation' },
+            reference,
+            true
+        );
+        const context: TurnContext = this.createContext(request);
+        return this.runMiddleware(context, logic);
+    }
+
+    public async processActivity(req: WebRequest, res: WebResponse, logic: (context: TurnContext) => Promise<any>): Promise<void> {
+
+        const message = await retrieveBody(req);
+
+        if (message == null) {
+            res.status(400);
+            res.end();
+        }
+
+        const activity: Partial<Activity> = {
+            id: message.id,
+            timestamp: new Date(),
+            channelId: this.channel,
+            conversation: {
+                id: message.id,
+                isGroup: false,
+                conversationType: null,
+                tenantId: null,
+                name: ''
+            },
+            from: {
+                id: message.user.id,
+                name: message.user.screen_name
+            },
+            recipient: {
+                id: message.in_reply_to_screen_name,
+                name: message.in_reply_to_screen_name
+            },
+            text: message.text,
+            channelData: message,
+            localTimezone: null,
+            callerId: null,
+            serviceUrl: null,
+            listenFor: null,
+            label: message.id,
+            valueType: null,
+            type: null
+        };
+
+        if (activity.type === ActivityTypes.Message) {
+
+            if(message.entities.media != null && message.entities.media.length > 0) {
+                const media = message.entities.media;
+                activity.attachments = [];
+                for (const medium of media) {
+                    let contentType: string;
+                    /*
+                     * Likely need to do a little more parsing of the content/mime type.
+                     */
+                    switch(medium.type) {
+                        case 'photo':
+                            contentType = 'image/png';
+                            break;
+                        case 'video':
+                            contentType = 'video/mpeg';
+                            break;
+                        case 'animated_gif':
+                            contentType = 'image/gif';
+                            break;
+                        default:
+                            throw new Error(`Medium doesn't have an associated content type.`);
```

Comment:
Do we want to throw an error or keep `contentType` empty or as 'unknown' or something.

Reply:
Unsure.

Reply:
Removed throwing exception in default.